naga/src/proc/typifier.rs

```diff
@@ -607,6 +606,18 @@ impl<'a> ResolveContext<'a> {
                     };
                     TypeResolution::Value(inner)
                 }
+                crate::BinaryOperator::LogicalAnd | crate::BinaryOperator::LogicalOr => {
+                    // These accept scalars only.
+                    let bool = Ti::Scalar(crate::Scalar::BOOL);
+                    let ty = past(left)?.inner_with(types);
+                    if *ty == bool {
+                        TypeResolution::Value(bool)
+                    } else {
+                        return Err(ResolveError::IncompatibleOperands(format!(
+                            "{op:?}({ty:?}, _)"
+                        )));
```

Comment:
Now that the overload database has landed, see if you can't use `common::ForDebugWithTypes::for_debug` to get a nicer string out of this. I think it might be as simple as:

```suggestion
                        return Err(ResolveError::IncompatibleOperands(format!(
                            "{op:?}({:?}, _)",
                            ty.for_debug(types),
                        )));
```